### ISSUE TITLE
Using @CognitoUser just to get user info without authentication/authorization

### ISSUE THREAD

**Issue by jgilewski-kone** [Lokicoule/nestjs-cognito#1170]
Hi!

I have the following issue: let's assume I have a public route, and it will return some general message if the user is not logged in (no JWT present), but for authenticated users, I would like to customize the message.

The controller would look like this:

```
  @Get()
  getMessage(@CognitoUser("username") username?: string) {
    return "Hello " + (username ? username : "stranger")
  }
```

How can I achieve this? Right now, `@Authentication` or `@Authorization` is required, but then I cannot access the endpoint publicly.

**Comment by Lokicoule** [comment]
Hi @jgilewski-kone!

Thanks for suggesting this feature - it's a great idea!

Just released v1.2.0 with optional auth - exactly what you proposed! 

Check out the docs for all the details:
https://github.com/Lokicoule/nestjs-cognito/tree/main/packages/auth#publicroute

Quick example of what you can do now:
```typescript
@Controller('api')
@Authentication()  // 👈 Don't forget this!
export class AppController {
    @PublicRoute()
    @Get('welcome')
    welcomeUser(@CognitoUser() user?: User) {
        return user 
            ? `Hey ${user.username}!` 
            : "Hey guest!";
    }
}
```
I'd love to hear how it works for your use case!

**Comment by jgilewski-kone** [comment]
Hi @Lokicoule!

Thanks for the super quick solution!
I have time to check it, and it works almost perfectly :)
The one thing that is not perfect is if you call this public route with the JWT token provided, we should always authenticate it if the `@Authentication` decorator is present. If JWT is invalid, 401 should be thrown even for `@PublicRoute`.
I can even provide some random string for the JWT token, and the route is called.
I expect that if the JWT is provided, it will be authenticated, and the route will be called. If there is no JWT token the public route will be called anyway without user information.
What do you think?

**Comment by Lokicoule** [comment]
Hi @jgilewski-kone!

Thanks for your feedback! 

Here are two possible solutions:

1. Enhanced `@PublicRoute`:
```ts
@PublicRoute({ strict: true })
endpoint() { }
```
2. New `@OptionalAuth()`:

```ts
@OptionalAuth()
endpoint() { }
```

Both would handle:
- No token → Allow access
- Valid token → Allow access + provide user info
- Invalid token → Return 401

`@OptionalAuth()` might be more explicit about the intent, while `@PublicRoute({ strict: true })` keeps everything in one decorator.

Which approach feels clearer to you?

**Comment by jgilewski-kone** [comment]
Hmm... in general, if my route has `@Authentication` decorator set and the user gives me the wrong JWT token (expired, from a different issuer or just some random string), I would unauthorize him wherever it is set in `@PublicRoute`.
If the user provides JWT, we need to authenticate it. If not, and the `@PublicRoute` is set, I would pass the request as an unauthenticated user.
In general, if both `@Authentication` and `@PublicRoute` are set, `@Authentication` should take precedence.

**Comment by Lokicoule** [comment]
Thanks for the suggestion! I agree with your logic, it's fair and straightforward. I've implemented it exactly as you described.
No need to overcomplicate things! 👍

Released in @nestjs-cognito/auth@1.2.1

**Comment by jgilewski-kone** [comment]
It works perfectly, thanks!
